config/SMART_STU2.yml

```diff
@@ -0,0 +1,37 @@
+---
+api:
+  url: http://smart:5000/api/fhir/
+
+auth:
+  strategy: authorization_code
+  client_id: test-suite
+  # client_secret intentionally included
+  client_secret: demo-secret-s4s
+  scope: launch/patient patient/*.read offline_access
+  confidential_client: True
+  sign_in_steps:
+    - element: '#sign-in'
+      click: True
+  authorize_steps:
+    - element: "[data-patient-id=smart-1288992]"
+      optional: True
+      click: True
+      wait: 2
+    - element: "[data-slide=next]"
+      optional: True
+      click: True
+      wait: 2
+    - element: "[type=checkbox]"
+      optional: True
+      execute_script: |
+        Array.from(document.querySelectorAll('input.glyphbox')).forEach(function(el) { el.disabled=false; });
+        Array.from(document.querySelectorAll('input.glyphbox')).forEach(function(el) { el.checked=true; });
+        Array.from(document.querySelectorAll('button[data-slide=next]')).forEach(function(el) { el.disabled=false; });
+        Array.from(document.querySelectorAll('button[data-slide=next]'))[1].click();
+    - element: "#authorize"
+      optional: True
+      click: True
+      wait: 2
+
+version:
```

Comment:
Two thoughts on this config item:

1. I think "version" should be under the "api" namespace.
2. Is there any situation where a single API would be able to handle both DSTU2 and STU3? My point being, do we need version.tags, or would a simple `version: DSTU2` be sufficient?

Reply:
I think we want to cover a case where a single config file says "go here for dstu2-style auth + clinical resources, and there for stu3-style imaging resources". So maybe represent with something like:

```yaml
use_cases:
  security: r2
  clinical: r2
  imaging: r3

api:
  versions:
    r2:
      url: https://url-for-r2-server
      // and other config here if needed
    r3:
      url: https://url-for-r3-server
      // and other config here if needed
```

This is hairy, but I think it captures the "essential complexity" that we want. We could then tag specific features with use case tags like `security`, `clinical`, `imaging` or `financial`... The logic above would mean:

* Any feature tagged with `use_case_security` must also be tagged with `r2`, otherwise we skip it
* Any feature tagged with `use_case_clinical` must also be tagged with `r2`, otherwise we skip it
* Any feature tagged with `use_case_imaging` must also be tagged with `r3`, otherwise we skip it
* Any feature tagged with a `use_case_*` not listed here gets skipped (e.g. `use_case_financial`)

Reply:
This reminds me of the "target features, not browsers" advice from the dark days of JS browser compatibility. A list of supported features/use cases makes more sense than a list of supported versions.

One thing I'd be worried about is that I'd give a pretty high degree of likelihood to `https://url-for-r2-server` and `https://url-for-r3-server` requiring different auth credentials, if not entirely different login paths. At that point, they'd need to be broken into separate config files anyways. Why not just start there, instead of having multiple versions of an API inside a single config file?

Reply:
Great point @erikwiffin. Let me clarify!

The goal here is to test a complete "Deployment scenario". So if, let's say, Cerner introduce a DSTU3 and STU3 server that were both available, we'd have *two* Cerner config files, like

## Config for DSTU2 Clinical server
```yaml
use_cases:
  security: r2
  clinical: r2

api:
  versions:
    r2:
      url: https://url-for-r2-server
      // and other config here if needed
```


## Config for STU3 Clinical server
```yaml 
use_cases:
  security: r3
  clinical: r3
api:
  versions:
    r3:
      url: https://url-for-r3-server
      // and other config here if needed
```

The only time you'd need to do what I described above (with mixed content in a single file) would be a split scenario as we're doing for imaging, where you'll go to a DSTU2 auth + clinical server, and a STU3 imaging server (presenting the token you got from the DSUT2-focused auth server).